src/middlewared/middlewared/plugins/system.py

```diff
@@ -98,6 +101,21 @@ class Config:
                         'Serial port specified has not been identified by the system'
                     )
 
+        syslog_server = data.get('syslogserver')
+        if syslog_server:
+            match = re.match(r"^[\w\.\-]+(\:\d+)?$", syslog_server)
+            if not match:
+                verrors.add(
+                    f'{schema}.syslogserver',
+                    'Invalid syslog server format'
+                )
+            elif ':' in syslog_server:
+                port = int(syslog_server.split(':')[-1])
+                if port < 0 or port > 65535:
+                    verrors.add(
+                        f'{schema}.syslogserver',
+                        'Port specified should be between 0 - 65535'
```

Comment:
"Should" is a recommendation, use "must" when it is required.
```suggestion
                        'Port must be in the range of 0 to 65535.'
```